mcs/class/Mono.Security/Mono.Security.Cryptography/KeyPairPersistence.cs

```diff
@@ -275,54 +275,61 @@ public void Remove ()
 
 #if INSIDE_CORLIB
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal static extern bool _CanSecure (string root);
+		unsafe internal static extern bool _CanSecure (char* root);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal static extern bool _ProtectUser (string path);
+		unsafe internal static extern bool _ProtectUser (char* path);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal static extern bool _ProtectMachine (string path);
+		unsafe internal static extern bool _ProtectMachine (char* path);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal static extern bool _IsUserProtected (string path);
+		unsafe internal static extern bool _IsUserProtected (char* path);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal static extern bool _IsMachineProtected (string path);
+		unsafe internal static extern bool _IsMachineProtected (char* path);
 #else
 		// Mono.Security.dll assembly can't use the internal 
 		// call (and still run with other runtimes)
 
 		// Note: Class is only available in Mono.Security.dll as
 		// a management helper (e.g. build a GUI app)
 
-		internal static bool _CanSecure (string root) 
+		unsafe internal static bool _CanSecure (char* root)
 		{
 			return true;
 		}
 
-		internal static bool _ProtectUser (string path)
+		unsafe internal static bool _ProtectUser (char* path)
 		{
 			return true;
 		}
 
-		internal static bool _ProtectMachine (string path)
+		unsafe internal static bool _ProtectMachine (char* path)
 		{
 			return true;
 		}
 
-		internal static bool _IsUserProtected (string path)
+		unsafe internal static bool _IsUserProtected (char* path)
 		{
 			return true;
 		}
 
-		internal static bool _IsMachineProtected (string path)
+		unsafe internal static bool _IsMachineProtected (char* path)
 		{
 			return true;
 		}
 #endif
 		// private stuff
 
-		private static bool CanSecure (string path) 
+		static void CheckString (string name, string value)
```

Comment:
This is not the first occurrence of this, please move it to corlib/Mono as a new extension type

Reply:
Use C# extension methods https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods